### ISSUE TITLE
FB.GetAppLink returns the previous deep link after focus change

### ISSUE THREAD

**Issue by gabetoth-dgh** [facebook/facebook-sdk-for-unity#190]
FB.GetAppLink returns the previous deep link after focus change, although there was no deep link app launch.

Reproduction:
1. Open app with deep link
2. call FB.GetAppLink(function) after FB init completes
3. put app to background
4. bring app back in focus
5. call FB.GetAppLink(function) again on focus change
6. the previous deep link is still present and the method above will return the same url.

Expected: FB.GetAppLink(function) is returning a deep link url only if the app was started/ brought in focus with a deep link.

**Comment by Jordan2507** [comment]
Noticed this issue in 7.10.0. Appears to only occur on Android. A decent workaround is to only process new deeplinks by storing the URL in implementation code and only processing the deeplink if it changes. This has a side effect of breaking the legitimate case of a user deeplinking, backgrounding, then deeplinking again.

**Comment by adam-govan** [comment]
This is still happening with **7.14.0** as well. Just tried it there and getting the same issue with Android only.

**Comment by tayl0r** [comment]
This is a pretty serious bug. When can we expect a fix, FB? Are you working on it?

**Comment by frankinshtein** [comment]
Android. happening with 7.17.2

**Comment by aisUnity** [comment]
HI gabetoth-dgh
i'm same issue Noticed please help me.
my code 
-------
void Awake()
    {
        if (!FB.IsInitialized)
        {
            FB.Init(InitCallback, OnHideUnity);
        }
        else
        {
            // Already initialized, signal an app activation App Event
            FB.ActivateApp();
        }
    }

private void InitCallback()
    {
        if (FB.IsInitialized)
        {
            FB.ActivateApp();
           FB.GetAppLink(DeepLinkCallback);
        }
        else
        {
            LogMessage.Log("Failed to Initialize the Facebook SDK");
        }
    }

private void DeepLinkCallback(IAppLinkResult result)
    {
        Debug.Log("DeepLink Response : " + result.Url);
    }

//background to reopen app and OnApplicationFocus code hear

void OnApplicationFocus(bool hasFocus)
    {
        if (hasFocus)
        {
            if (UIManager.Instance.isLoggedIn || UIManager.Instance.isGuestLogin)
            {
                Debug.Log("OnApplicationFocus");
                FB.GetAppLink(DeepLinkCallback);
            }
        }
    }


private void DeepLinkCallback(IAppLinkResult result)
    {
        Debug.Log("DeepLink Response : " + result.Url);
    }

- the previous deep link is still present and the method above will return the same url. same issue

**Comment by gabetoth-dgh** [comment]
@aisUnity workaround is to store the previous link that was opened, and if it matches the new one, do nothing.

**Comment by aisUnity** [comment]
@gabetoth-dgh  thank you for the reply  but my one more question how it can possible FB.GetAppLink response clear or null

**Comment by gabetoth-dgh** [comment]
@aisUnity I don't quite understand. Do You wan't to clear the response arrived in the delegate callback?

**Comment by aisUnity** [comment]
Yes, kind of. Once i have received the response from FB.GetAppLink, i am storing it in a local variable and want to clear the response object as whenever app is going in background and then come to foreground/focus, i am again getting that response which was already processed.

**Comment by gabetoth-dgh** [comment]
@aisUnity this is kind of the problem, that it get's stuck there.

**Comment by aisUnity** [comment]
@gabetoth-dgh I have one more question. If my game is already running and I am going to another app and click to deep-link of the game. Then it opens my game. But I am not getting any callback response from FB.GetAppLink. So Is there any other callback for that or anything else?

**Comment by roopesh-verma** [comment]
Has this issue been resolved or it still persists? As in my app I open the game using the deeplink and the process happens. When I keep the app in background and get it back on resume, it still uses the previously stored applink.

In FB.cs class there is a **ClearAppLink** method, but it has no implementation, is there any workaround to clear all the AppLinks previously generated?
